apps/OpenSpace/ext/launcher/src/profile/horizonsdialog.cpp

```diff
@@ -0,0 +1,1173 @@
+        );
+        end = QDateTime::fromString(
+            QString::fromStdString(_validTimeRange.second),
+            "yyyy-MMM-dd T hh:mm:ss"
+        );
+
+        if (!start.isValid() || !end.isValid()) {
+            QDate startDate = QDate::fromString(
+                QString::fromStdString(_validTimeRange.first),
+                "yyyy-MMM-dd"
+            );
+            QDate endDate = QDate::fromString(
+                QString::fromStdString(_validTimeRange.second),
+                "yyyy-MMM-dd"
+            );
+
+            if (startDate.isValid() && endDate.isValid()) {
+                _startEdit->setDate(startDate);
+                _endEdit->setDate(endDate);
+                _importTimeButton->hide();
+                _validTimeRange = std::pair<std::string, std::string>();
+                return;
+            }
+
+            _errorMsg->setText("Could not import time range");
+            appendLog(fmt::format("Could not import time range '{}' to '{}'",
+                _validTimeRange.first, _validTimeRange.second), LogLevel::Error
+            );
+            return;
+        }
+    }
+    _startEdit->setDateTime(start);
+    _endEdit->setDateTime(end);
+    _importTimeButton->hide();
+    _validTimeRange = std::pair<std::string, std::string>();
+}
+
+#ifdef OPENSPACE_MODULE_SPACE_ENABLED
+// When the user presses the 'Save' button
+void HorizonsDialog::approved() {
+    _downloadLabel->show();
+    bool result = handleRequest();
+    _downloadLabel->hide();
+    if (result && std::filesystem::is_regular_file(_horizonsFile.file())) {
+        accept();
+    }
+}
+#endif // OPENSPACE_MODULE_SPACE_ENABLED
+
+void HorizonsDialog::createWidgets() {
+    QGridLayout* layout = new QGridLayout(this);
+    layout->setSpacing(10);
+    {
+        QLabel* generateLabel = new QLabel("Generate a Horizons file", this);
+        generateLabel->setObjectName("heading");
+        layout->addWidget(generateLabel, 0, 0, 1, 3);
+
+        QLabel* infoLabel = new QLabel("<p>For more information about the Horizons "
+            "system please visit: <a href=\"https://ssd.jpl.nasa.gov/horizons/\">"
+            "https://ssd.jpl.nasa.gov/horizons/</a></p>",
+            this
+        );
+        infoLabel->setWordWrap(true);
+        infoLabel->setObjectName("thin");
+        infoLabel->setOpenExternalLinks(true);
+        layout->addWidget(infoLabel, 1, 0, 1, 3);
+    }
+    {
+        _typeLabel = new QLabel("Horizons data type:", this);
+        _typeLabel->setToolTip("Choose Horizons data type");
+        layout->addWidget(_typeLabel, 2, 0, 1, 2);
+
+        _typeCombo = new QComboBox(this);
+        _typeCombo->setToolTip("Choose Horizons data type");
+        QStringList types = {
+           "Vector table",
+           "Observer table"
+        };
+        _typeCombo->addItems(types);
+        _typeCombo->setCurrentIndex(0);
+        connect(_typeCombo, QOverload<int>::of(&QComboBox::currentIndexChanged),
```

Comment:
According to Alex the guidelines are more relaxed when it comes to the Qt connect function. He commented the exact opposite since I used to have it like you are suggesting.

Reply:
OK, nvm then :)